### PR TITLE
More SonarCloud "bug" fixes in Imath

### PR DESCRIPTION
**Pull request by xlietz**
Various SonarCloud bug fixes:
- added explicit destructors.
- added explicit copy constructor.
- added NOSONAR exceptions for array index operators in math primitive classes, current method of accessing is proven and fast.
- removed unreachable return statement.

### CODE REVIEW

File: IlmBase/Imath/ImathEuler.h [hunk 1]
```diff
@@ -226,6 +226,12 @@ class Euler : public Vec3<T>
     Euler(const Matrix33<T> &, Order o = Default);
     Euler(const Matrix44<T> &, Order o = Default);
 
+    //-------------
+    //  Destructor
+    //-------------
+	
+	~Euler();
```
Comment (cary-ilm) [inline IlmBase/Imath/ImathEuler.h]: We should be able to declare this as ~Euler() = default; and not have to provide the definition below. That should force it to be the empty function - same behavior but fewer lines of code, and the intention is a bit clearer. Same with the other classes below.

Comment (xlietz) [reply IlmBase/Imath/ImathEuler.h]: Sounds good, I will change that everywhere!

Comment (xlietz) [reply IlmBase/Imath/ImathEuler.h]: I pushed a new version, let me know if all looks good now!

**Review by cary-ilm — APPROVED** [review]
LGTM

**Pull request merged**